tracing/src/main/java/com/palantir/remoting1/tracing/AsyncSlf4jSpanObserver.java

```diff
@@ -32,34 +38,63 @@
  * with log-level {@link Logger#info INFO}. Logging is performed asynchronously on a given executor service.
  */
 public final class AsyncSlf4jSpanObserver extends AsyncSpanObserver {
+    private static final ObjectMapper mapper = new ObjectMapper().registerModule(new GuavaModule());
 
     private final Logger logger;
+    private final ZipkinCompatEndpoint endpoint;
 
-    @JsonDeserialize(as = ImmutableZipkinCompatibleSerializableSpan.class)
-    @JsonSerialize(as = ImmutableZipkinCompatibleSerializableSpan.class)
+    @JsonSerialize(as = ImmutableZipkinCompatSpan.class)
     @Value.Immutable
     @Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE)
-    abstract static class ZipkinCompatibleSerializableSpan {
-        private static final ObjectMapper mapper = new ObjectMapper().registerModule(new GuavaModule());
+    abstract static class ZipkinCompatSpan {
 
         abstract String getTraceId();
         abstract String getId();
         abstract String getName();
         abstract Optional<String> getParentId();
         abstract long getTimestamp();
         abstract long getDuration();
+        abstract List<ZipkinCompatAnnotation> annotations();
 
-        static ZipkinCompatibleSerializableSpan fromSpan(Span span) {
-            return ImmutableZipkinCompatibleSerializableSpan.builder()
+        static ZipkinCompatSpan fromSpan(Span span, ZipkinCompatEndpoint endpoint) {
+            return ImmutableZipkinCompatSpan.builder()
                     .traceId(span.getTraceId())
                     .id(span.getSpanId())
                     .name(span.getOperation())
                     .parentId(span.getParentSpanId())
                     .timestamp(span.getStartTimeMicroSeconds())
                     .duration(nanoToMicro(span.getDurationNanoSeconds()))  // Zipkin-durations are micro-seconds, round
+                    .addAllAnnotations(spanTypeToZipkinAnnotations(span, endpoint))
                     .build();
         }
 
+        private static Iterable<? extends ZipkinCompatAnnotation> spanTypeToZipkinAnnotations(
+                Span span, ZipkinCompatEndpoint endpoint) {
+            if (!span.type().isPresent()) {
+                return ImmutableList.of();
+            }
+
+            List<ZipkinCompatAnnotation> annotations = Lists.newArrayListWithCapacity(2);
+            switch (span.type().get()) {
+                case CLIENT_OUTGOING:
+                    annotations.add(ZipkinCompatAnnotation.of(span.getStartTimeMicroSeconds(), "cs", endpoint));
+                    annotations.add(ZipkinCompatAnnotation.of(
+                            span.getStartTimeMicroSeconds() + nanoToMicro(span.getDurationNanoSeconds()),
+                            "cr",
+                            endpoint));
```

Comment:
Do we want to capture annotations marking errors?

See https://github.com/openzipkin/zipkin/blob/master/zipkin/src/main/java/zipkin/Constants.java